olive/passes/onnx/extract_adapters.py

```diff
@@ -350,3 +373,23 @@ def _create_empty_initializer(cls, dag: OnnxDAG, weights: Dict[str, "NDArray"],
         new_initializer = cls._copy_initializer(old_proto, new_name)
         # add the new initializer to the graph
         dag.add_initializer(new_initializer, dag.get_io(old_name).graph_idx)
+
+    @classmethod
+    def _make_dynamic_optional(cls, dag: OnnxDAG, weights: Dict[str, "NDArray"], name: str, config: Dict[str, Any]):
+        """Make the input dynamic and optional."""
+        if "quant" in name:
+            # dynamic shape and optional inputs not supported for quantized modules yet
+            return
+
+        dim_idx = 1 if "lora_A" in name else 0
```

Comment:
Do we have to hard code "lora_A"?

Reply:
the lora r dimension index depends on whether it's the A matrix or B matrix, so we need this information. The names here are after they have already been standardized https://github.com/microsoft/Olive/blob/056741839f1c0f8b7f9b7c1785b928c8781f37d7/olive/passes/onnx/extract_adapters.py#L319 so lora_A or lora_B is already present.